src/pyfmi/fmi.pyx

```diff
@@ -4087,26 +4110,26 @@ cdef class FMUModelBase2(ModelBase):
         if self._fmu_kind == FMIL.fmi2_fmu_kind_unknown:
             last_error = decode(FMIL.jm_get_last_error(&self.callbacks))
             if enable_logging:
-                raise FMUException("The FMU kind could not be determined. "+last_error)
+                raise InvalidVersionException("The FMU kind could not be determined. "+last_error)
             else:
-                raise FMUException("The FMU kind could not be determined. Enable logging for possibly more information.")
+                raise InvalidVersionException("The FMU kind could not be determined. Enable logging for possibly more information.")
         elif self._fmu_kind == FMIL.fmi2_fmu_kind_me_and_cs:
             if isinstance(self,FMUModelME2):
                 self._fmu_kind = FMIL.fmi2_fmu_kind_me
             elif isinstance(self,FMUModelCS2):
                 self._fmu_kind = FMIL.fmi2_fmu_kind_cs
             else:
-                raise FMUException("FMUModelBase2 cannot be used directly, use FMUModelME2 or FMUModelCS2.")
+                raise InvalidVersionException("FMUModelBase2 cannot be used directly, use FMUModelME2 or FMUModelCS2.")
```

Comment:
I'm not convinced this should be an InvalidVersionException, version is 2 in all three cases here.

Reply:
I agree, it is not ideal. Any suggestion?

Reply:
For now, I suggest to keep the FMUException.